src/karpatkit/functions.py

```diff
@@ -51,8 +52,10 @@ def to_token_amount(
     Returns:
         Decimal: The converted token amount.
     """
-    decimals = get_decimals(token_address, blockchain=blockchain, web3=web3) if decimals else 0
-    return amount / Decimal(10**decimals)
+    if decimals:
+        decs = get_decimals(token_address, blockchain=blockchain, web3=web3)
+        return amount / Decimal(10**decs)
```

Comment:
what if we get no decimals?

Reply:
it didn't handle that scenario before either right?